vendor/k8s.io/kubernetes/openshift-kube-apiserver/admission/authorization/restrictusers/usercache/groups.go

```diff
@@ -31,10 +34,17 @@ func ByUserIndexKeys(obj interface{}) ([]string, error) {
 func NewGroupCache(groupInformer userinformer.GroupInformer) *GroupCache {
 	return &GroupCache{
 		indexer: groupInformer.Informer().GetIndexer(),
+		groupsSynced: groupInformer.Informer().HasSynced,
 	}
 }
 
 func (c *GroupCache) GroupsFor(username string) ([]*userapi.Group, error) {
+	err := wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
```

Comment:
wrong spot. You can track the `groupsSynced` in this struct, but leave the handling up to the callers of this method.  I doubt they will be uniform.  Admission handlers may choose to block on partial lists and then fail. Authentication is likely to block and then make the best choice it can.

Reply:
@deads2k , please review update. I also have https://github.com/openshift/origin/pull/24414 for the authentication use of this utility.